XPlot.DotNet.Interactive.KernelExtensions/XPlotKernelExtension.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.DotNet.Interactive;
+using Microsoft.DotNet.Interactive.Rendering;
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using XPlot.Plotly;
+using static Microsoft.DotNet.Interactive.Rendering.PocketViewTags;
+
+namespace XPlot.DotNet.Interactive.KernelExtensions
+{
+    public class XPlotKernelExtension : IKernelExtension
+    {
+        public Task OnLoadAsync(IKernel kernel)
+        {
+            Formatter<PlotlyChart>.Register((chart, writer) =>
+           {
+               writer.Write(GetChartHtml(chart));
+           }, "text/html");
+
+            return Task.CompletedTask;
+        }
+
+        private string GetChartHtml(PlotlyChart chart)
+        {
+            string chartHtml = chart.GetInlineHtml();
+
+            int scriptStart = chartHtml.IndexOf("<script>") + "<script>".Length;
+            int scriptEnd = chartHtml.IndexOf("</script>");
+
+            StringBuilder html = new StringBuilder(chartHtml.Length);
+            html.Append(chartHtml,0, scriptStart);
+
+            html.Append(@"
+require.config({paths:{plotly:'https://cdn.plot.ly/plotly-latest.min'}});
+require(['plotly'], function(Plotly) { 
+");
+
+            html.Append(chartHtml, scriptStart + 1, scriptEnd - scriptStart - 1);
+
+            html.AppendLine(@"});");
+
+            html.Append(chartHtml.AsSpan().Slice(scriptEnd).ToString());
```

Comment:
You should use the same StringBuilder overloads as you have above - (string, start, length), and not create a span and call ToString on it, since that creates and copies unnecessary memory.

Reply:
Even simpler is to pass in the `TextWriter` from the `Formatter.Register` call on line 15.

Reply:
Even better. In that case, I'd suggest making this project target `netcoreapp2.1` or higher (or `netstandard2.1`), and then take advantage of the new `ReadOnlySpan` overload - https://docs.microsoft.com/en-us/dotnet/api/system.io.textwriter.write?view=netcore-2.1#System_IO_TextWriter_Write_System_ReadOnlySpan_System_Char__.

I'm sure it won't make a dramatic different performance wise, but why make the computer do unnecessary work when it doesn't have to.

Reply:
Instead of parsing strings, I have modified the code to use the HTMLAgilityPack